### PR TITLE
example pr for rabit cache

### PR DESCRIPTION
**Pull request by chenqin**

### CODE REVIEW

**Comment by thvasilo** [comment]
Hey @chenqin, trying to check out this PR I get the following errors `Server does not allow request for unadvertised object`:

```
git fetch origin pull/4636/head:chenqin
remote: Enumerating objects: 25, done.
remote: Counting objects: 100% (25/25), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 36 (delta 23), reused 24 (delta 23), pack-reused 11
Unpacking objects: 100% (36/36), done.
From https://github.com/dmlc/xgboost
 * [new ref]           refs/pull/4636/head -> chenqin
Fetching submodule dmlc-core
From https://github.com/dmlc/dmlc-core
   d06aceb..f1ff6cc  master     -> origin/master
Fetching submodule rabit
error: Server does not allow request for unadvertised object db343f9394470d63bc2c95551d3beb80f39086c3
error: Server does not allow request for unadvertised object ee234c4bb05422963491cae35f391d1e15d88f29
error: Server does not allow request for unadvertised object ef235c4308f64170d586743c6d0897239a7a55be
error: Server does not allow request for unadvertised object 7186da620fa4897545c34d8b7bb0a40cd8001e65
error: Server does not allow request for unadvertised object f32edca5812b1b5160c15ea2c40034048f5af843

```

File: src/tree/updater_quantile_hist.cc [hunk 1]
```diff
@@ -223,6 +223,9 @@ void QuantileHistMaker::Builder::ExpandWithDepthWidth(
     int sync_count = 0;
     std::vector<ExpandEntry> temp_qexpand_depth;
     BuildLocalHistograms(&starting_index, &sync_count, gmat, gmatb, p_tree, gpair_h);
+    //TODO (chen qin): remove me once root caused, force sync count in sync
+    rabit::Allreduce<rabit::op::Min>(&sync_count, sizeof(int));
```
Comment (CodingCat) [inline src/tree/updater_quantile_hist.cc]: obviously this is something should not be here

Comment (chenqin) [reply src/tree/updater_quantile_hist.cc]: removed

**Comment by chenqin** [comment]
> Hey @chenqin, trying to check out this PR I get the following errors `Server does not allow request for unadvertised object` when checking out rabit:
> 
> ```
> git fetch origin pull/4636/head:chenqin
> remote: Enumerating objects: 25, done.
> remote: Counting objects: 100% (25/25), done.
> remote: Compressing objects: 100% (2/2), done.
> remote: Total 36 (delta 23), reused 24 (delta 23), pack-reused 11
> Unpacking objects: 100% (36/36), done.
> From https://github.com/dmlc/xgboost
>  * [new ref]           refs/pull/4636/head -> chenqin
> Fetching submodule dmlc-core
> From https://github.com/dmlc/dmlc-core
>    d06aceb..f1ff6cc  master     -> origin/master
> Fetching submodule rabit
> error: Server does not allow request for unadvertised object db343f9394470d63bc2c95551d3beb80f39086c3
> error: Server does not allow request for unadvertised object ee234c4bb05422963491cae35f391d1e15d88f29
> error: Server does not allow request for unadvertised object ef235c4308f64170d586743c6d0897239a7a55be
> error: Server does not allow request for unadvertised object 7186da620fa4897545c34d8b7bb0a40cd8001e65
> error: Server does not allow request for unadvertised object f32edca5812b1b5160c15ea2c40034048f5af843
> ```

sorry about late reply can you give another try against? https://github.com/chenqin/xgboost/commit/7f5f278b3c87f2fc143db7006cd262e5c9333c8d

File: src/tree/updater_quantile_hist.cc (hunk 1 above)
Comment (CodingCat) [reply src/tree/updater_quantile_hist.cc]: it's still here?

**Comment by CodingCat** [comment]
I just found this PR may also involve conflict with the issue at https://github.com/dmlc/xgboost/pull/4594 (where more rabit operations are to be added )

**Comment by chenqin** [comment]
> I just found this PR may also involve conflict with the issue at #4594 (where more rabit operations are to be added )

Let's review design document first once it's out.

**Pull request closed without merging**